mod/01_filter/set/right/filter_search_metric.rb

```diff
@@ -0,0 +1,96 @@
+include_set Abstract::Filter
+
+def default_sort_by_key
+  "upvoted"
+end
+
+def shift_sort_table?
+  %w(values company).include?(Env.params["sort"] || default_sort_by_key)
+end
+
+def params_keys
+  %w(metric designer wikirate_topic project year)
+end
+
+def search_wql opts, return_param=nil
+  wql = { type_id: MetricID }
+  wql[:return] = return_param if return_param
+  filter_by_name wql, opts[:metric]
+  filter_by_designer wql, opts[:designer]
+  filter_by_topic wql, opts[:wikirate_topic]
+  filter_by_year wql, opts[:year]
+  filter_by_project wql, opts[:project]
+  wql
+end
```

Comment:
They look all similar these `search_wql` methods.
Use `params_keys.each { |key| send("filter_by_#{key}", wql, opts[key] }` and move it to Abstract::Filter?